### PR TITLE
[Oracle] Add logs

### PR DESCRIPTION
**Pull request by moxarth-elastic**
## Part Of https://github.com/elastic/connectors/issues/2299

Adding more logs in Oracle connector.

Log file: https://drive.google.com/file/d/1eqC6FRfm4IicdMNt2Z1Hivw6x3G-g_JC/view?usp=drive_link

#### Pre-Review Checklist
- [x] this PR does NOT contain credentials of any kind, such as API keys or username/passwords (double check `config.yml.example`)
- [x] this PR has a meaningful title
- [x] this PR links to all relevant github issues that it fixes or partially addresses
- [x] if there is no GH issue, please create it. Each PR should have a link to an issue
- [x] this PR has a thorough description
- [x] Tested the changes locally
- [x] Added a label for each target release version (example: `v7.13.2`, `v7.14.0`, `v8.0.0`)

### CODE REVIEW

**Comment by moxarth-elastic** [comment]
buildkite test this

**Comment by artem-shelkovnikov** [comment]
@moxarth-elastic example log file that you've attached is actually for OneDrive, can you provide proper log file?

**Comment by artem-shelkovnikov** [comment]
The most useful connector logs are - which queries were actually executed by the connectors. Can you add debug logs that state which exact queries were executed? A good addition to it would be to also debug log the number of records returned by the query.

**Comment by moxarth-elastic** [comment]
> @moxarth-elastic example log file that you've attached is actually for OneDrive, can you provide proper log file?

Oops! My bad. I've updated the link.

File: connectors/sources/oracle.py [hunk 1]
```diff
@@ -242,6 +250,8 @@ async def data_streamer(self, table):
         Yields:
             list: It will first yield the column names, then data in each row
         """
+        self._logger.debug(f"Streaming records from database for table: {table}")
```
Comment (artem-shelkovnikov) [inline connectors/sources/oracle.py:253]:
```suggestion
        self._logger.debug(f"Streaming records from database for table \"{table}\"")
```

File: connectors/sources/oracle.py [hunk 2]
```diff
@@ -253,7 +263,9 @@ async def data_streamer(self, table):
             fetch_size=self.fetch_size,
             retry_count=self.retry_count,
         ):
+            record_count += 1
             yield data
+        self._logger.info(f"Found {record_count} records from table: {table}")
```
Comment (artem-shelkovnikov) [inline connectors/sources/oracle.py:268]:
```suggestion
        self._logger.info(f"Found {record_count} records in table  \"{table}\"")
```

File: connectors/sources/oracle.py [hunk 3]
```diff
@@ -423,9 +435,11 @@ async def fetch_documents(self, table):
             Dict: Document to be indexed
         """
         try:
+            self._logger.info(f"Fetching records for the table: {table}")
```
Comment (artem-shelkovnikov) [inline connectors/sources/oracle.py:438]:
```suggestion
            self._logger.info(f"Fetching records for the table \"{table}\"")
```

File: connectors/sources/oracle.py [hunk 4]
```diff
@@ -423,9 +435,11 @@ async def fetch_documents(self, table):
...
             row_count = await self.oracle_client.get_table_row_count(table=table)
             if row_count > 0:
                 # Query to get the table's primary key
+                self._logger.debug(f"Total '{row_count}' rows found in '{table}' table")
```
Comment (artem-shelkovnikov) [inline connectors/sources/oracle.py:442]:
```suggestion
                self._logger.debug(f"Total {row_count} rows found in table \"{table}\"")
```

Comment (moxarth-elastic) [reply connectors/sources/oracle.py]: @artem-shelkovnikov these changes are causing the linting error.

Comment (artem-shelkovnikov) [reply connectors/sources/oracle.py]: It's not automatically fixable by `make autoformat`?

Comment (moxarth-elastic) [reply connectors/sources/oracle.py]: I already tried to do that but it was reformatting the log to the previous one. It's not accepting the current logs you suggested.

Comment (artem-shelkovnikov) [reply connectors/sources/oracle.py]: It cannot reformat to previous one, because I've changed wording of the log slightly?

Comment (artem-shelkovnikov) [reply connectors/sources/oracle.py]: Does it complain at `\"`?

Comment (moxarth-elastic) [reply connectors/sources/oracle.py]: Yes, it removes the escaping

Comment (artem-shelkovnikov) [reply connectors/sources/oracle.py]: We can do that, it's okay, as long as it's consistent.

**Comment by artem-shelkovnikov** [comment]
@moxarth-elastic why did you revert the whole change? The only problem was the escaping? Now I see that log lines are inconsistent - some say`table: {table}`, some say `'{table}' table`.

**Comment by moxarth-elastic** [comment]
> @moxarth-elastic why did you revert the whole change? The only problem was the escaping? Now I see that log lines are inconsistent - some say`table: {table}`, some say `'{table}' table`.

Oh, I see. Let me update it.

**Comment by moxarth-elastic** [comment]
@artem-shelkovnikov I've made the logs consistent and fixed the linter error.

**Comment by navarone-feekery** [comment]
This is ready for review (again)

**Review by seanstory — APPROVED** [review]

**Comment by navarone-feekery** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.x|[](https://github.com/elastic/connectors/pull/2865)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**